src/gettingstarted/own-code/project-configuration.md

```diff
@@ -0,0 +1,87 @@
+        makeButton(descButtonPython);
+
+        var descPathNode = getPathObj("/languages/nodejs.html#project-templates", "Node.js");
+        var descButtonNode = {type: "basicFull", path: descPathNode, div: "nodejs"};
+        makeButton(descButtonNode);
+
+        var descPathRuby = getPathObj("/languages/ruby.html#project-templates", "Ruby");
+        var descButtonRuby = {type: "basicFull", path: descPathRuby, div: "ruby"};
+        makeButton(descButtonRuby);
+
+        var descPathGo = getPathObj("/languages/go.html#project-templates", "Go");
+        var descButtonGo = {type: "basicFull", path: descPathGo, div: "go"};
+        makeButton(descButtonGo);
+
+        var descPathJava = getPathObj("/languages/java.html#project-templates", "Java");
+        var descButtonJava = {type: "basicFull", path: descPathJava, div: "java"};
+        makeButton(descButtonJava);
+    </script>
+
+2. **Create empty configuration files**
+
+    You will notice that each of the templates above contain the following structure around their application code:
+
+    ```.
+    ├── .platform
+    │   ├── routes.yaml
+    │   └── services.yaml
+    ├── .platform.app.yaml
+    └── < application code >
+    ```
+
+    In order to successfully deploy to Platform.sh you must add three YAML files:
+
+      * A `.platform/routes.yaml` file, which configures the [routes](/configuration/routes.md) used in your environments. That is, it describes how an incoming HTTP request is going to be processed by Platform.sh.
+      * A `.platform/services.yaml` file, which configures the [services](/configuration/services.md) that will be used by the application. Connecting to Platform.sh's maintained services only requires properly writing this file. While this file must be present, if your application does not require services it can remain empty.
+      * At least one `.plaform.app.yaml` file, which configures the [application](/configuration/app-containers.md) itself. It provides control over the way the application will be built and deployed on Platform.sh.
+
+    When you set Platform.sh as a remote for your repository in the previous step, the CLI automatically created the hidden configuration directory `.platform` for you. The next steps will explore in more detail what each configuration files must include, but for now create empty files in their place.
+
+    ```bash
+    touch .platform/routes.yaml && touch .platform/services.yaml
+    touch .platform.app.yaml
+    ```
+
+With the empty configuration files in place, you will need to specify your service configuration in `.platform/services.yaml`.
+
+<div id = "buttons"></div>
+
+<script>
+$(document).ready(function(){
+  var navNextText = "I have initialized my code with empty configuration files";
+  var navButtons = {type: "navigation", prev: getPathObj("prev"), next: getPathObj("next", navNextText), div: "buttons"};
+  makeButton(navButtons);
+});
+</script>
```

Comment:
I so desperately want to replace this with a custom element, but browser support is still not-quite-there for Edge. :frowning_face:

Reply:
The only reason the buttons exists like this is to grab the element Gitbook generates `navigation navigation-prev/next` https://github.com/platformsh/platformsh-docs/blob/03206262613082e3212447647ec4dc8ddf86f637/src/scripts/buttons.js#L5 so that this can remain easy to maintain should files move around/get deleted/added. I'm up for any suggestions here though.